kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableSchemaEvolutionSuite.scala

```diff
@@ -59,17 +59,10 @@ class DeltaTableSchemaEvolutionSuite extends DeltaTableWriteSuiteBase with Colum
           fieldMetadataForColumn(3, "b"))
         .add("c", IntegerType.INTEGER, true, currentSchema.get("c").getMetadata)
 
-      table.createTransactionBuilder(engine, testEngineInfo, Operation.MANUAL_UPDATE)
-        .withSchema(engine, newSchema)
-        .build(engine).commit(engine, emptyIterable())
-
-      val structType = table.getLatestSnapshot(engine).getSchema
-      assertColumnMapping(structType.get("a"), 1)
+      updateTableMetadata(engine, tablePath, newSchema)
 
-      val innerStruct = structType.get("b").getDataType.asInstanceOf[StructType]
-      assertColumnMapping(innerStruct.get("d"), 4, "d")
-      assertColumnMapping(innerStruct.get("e"), 5, "e")
-      assertColumnMapping(structType.get("c"), 2)
+      val actualSchema = getMetadata(engine, tablePath).getSchema
+      assert(actualSchema == newSchema)
```

Comment:
I changed all the asserts to just do equality checks against the schema, `equals` verifies metadata like field IDs as well. It also makes sure we're not missing any assertions on fields that should not be there etc.